.github/workflows/valgrind_ut.yml

```diff
@@ -0,0 +1,176 @@
+#
+# If not stated otherwise in this file or this component's LICENSE file the
+# following copyright and licenses apply:
+#
+# Copyright 2023 Sky UK
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This workflow builds and runs the rialto-ocdm unittests with valgrind to check for memory leaks.
+# The workflow shall fail if leaks are detected and the results are summarized in github.
+# Logs are uploaded on failure.
+
+name: valgrind_ut
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "master" branch
+  push:
+    branches: [ "master", "rdkcentral:master" ]
+  pull_request:
+    branches: [ "master", "rdkcentral:master" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+
+  # This job downloads and caches valgrind to speed up execution
+  # An updated version of valgrind is required, the default version for ubuntu-18.04 is
```

Comment:
Update this comment, probably can remove this now and just do apt install valgrind for Ubuntu 22. We should then be able to remove all this caching for valgrind. Maybe try it out after delivering this.

Reply:
comment removed. I will try to install valgrind using apt-get in next commit